crates/ruff_linter/src/settings/types.rs

```diff
@@ -298,15 +331,12 @@ impl<T> PerFile<T> {
         if negated {
             pattern.drain(..1);
         }
-        let path = Path::new(&pattern);
-        let absolute = match project_root {
-            Some(project_root) => fs::normalize_path_to(path, project_root),
-            None => fs::normalize_path(path),
-        };
+
+        let project_root = project_root.unwrap_or(&path_dedot::CWD);
```

Comment:
Same here, let's avoid spreading `path_dedot::CWD` over our code base and instead use a helper to maximize WASM compatibility